### PR TITLE
Remove underscore from observe sequence

### PR DESCRIPTION
**Pull request by harryadel**
As I'm wrapping up https://github.com/meteor/meteor/pull/11869 [Test Group 3](https://app.circleci.com/pipelines/github/meteor/meteor/2782/workflows/150e8e89-1d4d-4bbd-b448-e05355b496c4/jobs/77335) is failing because it runs non-core packages and observe-sequence [requires underscore](https://github.com/meteor/blaze/blob/master/packages/observe-sequence/package.js#L20) which is no longer around since now it's now deprecated. There's no need for publishing a new version of Blaze or Observer-sequence; this PR just needs to be merged in dev branch so in meteor repo the latest commit can pulled. Coincidentally, I'm glad the the underscore removal was done here first as it'd not have worked if it had been done in Meteor repo first. :sweat_smile:

### CODE REVIEW

**Review by StorytellerCZ — APPROVED** [review]
LGTM

**Comment by StorytellerCZ** [comment]
Let's merge this into 2.6 release and release 2.6. What do you think @filipenevola ?

**Review by fredmaiaarantes — APPROVED** [review]
LGTM

File: packages/observe-sequence/observe_sequence_tests.js [hunk 1]
```diff
@@ -678,7 +678,7 @@ Tinytest.add('observe-sequence - vm generated number arrays', function (test) {
 });
 
 Tinytest.add('observe-sequence - number arrays, _id:0 correctly handled, no duplicate ids warning #4049', function (test) {
-  var seq = _.map(_.range(3), function (i) { return { _id: i}; });
+  var seq = [...Array(2).keys()].map(function (i) { return { _id: i}; });
```
Comment (renanccastro) [inline packages/observe-sequence/observe_sequence_tests.js:681]: they are not equivalent, you are missing the third element here

File: packages/observe-sequence/observe_sequence_tests.js [hunk 2]
```diff
@@ -45,11 +45,11 @@ runOneObserveSequenceTestCase = function (test, sequenceFunc,
...
       firedCallbacks.splice(i, 0, obj);
     },
-    removedAt: function () {
-      firedCallbacks.push({removedAt: _.toArray(arguments)});
+    removedAt: function (...args) {
+      firedCallbacks.push({removedAt: args});
     },
     movedTo: function () {
-      firedCallbacks.push({movedTo: _.toArray(arguments)});
+      firedCallbacks.push({movedTo: args});
```
Comment (renanccastro) [inline packages/observe-sequence/observe_sequence_tests.js:52]: there is no mention to args in this function parameters

**Review by renanccastro — CHANGES_REQUESTED** [review]

File: packages/observe-sequence/observe_sequence_tests.js (hunk 2 above)
Comment (harryadel) [reply packages/observe-sequence/observe_sequence_tests.js:52]: Thanks for noticing these @renanccastro. You may review again now.

**Comment by harryadel** [comment]
Any update on this and the 2.6 release? @StorytellerCZ @renanccastro @filipenevola

**Comment by StorytellerCZ** [comment]
@fredmaiaarantes @jankapunkt let's include this in 2.6?

**Review by jankapunkt — COMMENTED** [review]
I would generally add it to 2.6 but I'd like to discuss one more this PR (see my comment on the code).

If this is to be a showstopper (due to further discussion) then we might publish 2.6 with this PR as-is and move the final removal of lodash (see my comment on the code) to 3.0?

File: packages/observe-sequence/observe_sequence.js [hunk 3]
```diff
@@ -1,3 +1,6 @@
+const isObject = Npm.require('lodash.isobject');
+const has = Npm.require('lodash.has');
+
```
[the remark this answers is not in the document]
Comment (StorytellerCZ) [reply packages/observe-sequence/observe_sequence.js:3]: I'm for replacement. Less dependencies is better.

**Comment by StorytellerCZ** [comment]
@harryadel can you please address @jankapunkt comment? Then we can merge this into 2.6 and release.

**Comment by harryadel** [comment]
Right away!

**Comment by StorytellerCZ** [comment]
@harryadel this seems to be the issue of the failing tests:
```
TypeError: posNew.forEach is not a function
    at diffArray
```

**Comment by harryadel** [comment]
It's weird that some errors weren't caught by the CI before :thinking:

**Pull request merged**